pkg/upgrade/upgrades/alter_jobs_add_job_type.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package upgrades
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/systemschema"
+	"github.com/cockroachdb/cockroach/pkg/upgrade"
+)
+
+const addTypeColumnStmt = `
+ALTER table system.jobs
+ADD COLUMN IF NOT EXISTS type STRING
+FAMILY "fam_0_id_status_created_payload"
+`
+
+const addTypeColumnIdxStmt = `
+CREATE INDEX IF NOT EXISTS jobs_type_idx
+ON system.jobs (type)
+`
+
+func alterSystemJobsAddJobType(
+	ctx context.Context, cs clusterversion.ClusterVersion, d upgrade.TenantDeps,
+) error {
+	op := operation{
+		name:           "add-jobs-type-col",
+		schemaList:     []string{"type"},
+		query:          addTypeColumnStmt,
+		schemaExistsFn: hasColumn,
+	}
+	if err := migrateTable(ctx, cs, d, op, keys.JobsTableID, systemschema.JobsTable); err != nil {
+		return err
+	}
+
+	op = operation{
+		name:           "add-jobs-type-col-idx",
+		schemaList:     []string{"jobs_type_idx"},
+		query:          addTypeColumnIdxStmt,
+		schemaExistsFn: hasIndex,
+	}
+	if err := migrateTable(ctx, cs, d, op, keys.JobsTableID, systemschema.JobsTable); err != nil {
+		return err
+	}
```

Comment:
You could put these into a for loop so that you only need to write the code calling `migrateTable` once.